src/cdomains/intDomain.ml

```diff
@@ -923,6 +923,461 @@ struct
   let project ik p t = t
 end
 
+module IntervalSetFunctor(Ints_t : IntOps.IntOps): S with type int_t = Ints_t.t and type t = (Ints_t.t * Ints_t.t) list =
+struct
+
+  module Interval_functor = IntervalFunctor(Ints_t) 
+
+  let name () = "IntervalSets"
+
+  type int_t = Ints_t.t
+
+  type t = (Ints_t.t * Ints_t.t) list [@@deriving eq, hash, ord]
+
+  let range ik = BatTuple.Tuple2.mapn Ints_t.of_bigint (Size.range ik)
+
+  let top () = failwith @@ "top () not implemented for " ^ (name ())
+
+  let top_of ik = [range ik]
+   
+  let bot () = []
+  
+  let bot_of ik = bot () 
+
+  let show (x: t) =
+    let show_interval i = Printf.sprintf "[%s, %s]" (Ints_t.to_string (fst i)) (Ints_t.to_string (snd i)) in
+    List.fold_left (fun acc i -> (show_interval i) :: acc) [] x |> List.rev |> String.concat ", " |> Printf.sprintf "[%s]"
+  
+  let cartesian_product l1 l2 = 
```

Comment:
This function is used only once, please pull it down to where it is used.  Also, maybe `BatList.cartesian_product`?